lib/config/Configuration.js

```diff
@@ -11,13 +11,16 @@ import os from "node:os";
  */
 class Configuration {
 	#mavenSnapshotEndpointUrl;
+	#ui5DataDir;
 
 	/**
 	 * @param {object} configuration
 	 * @param {string} [configuration.mavenSnapshotEndpointUrl]
+	 * @param {string} [configuration.ui5DataDir]
 	 */
-	constructor({mavenSnapshotEndpointUrl}) {
+	constructor({mavenSnapshotEndpointUrl, ui5DataDir}) {
```

Comment:
currently the configuration does not offer a default value but in real the home dir has a default value.... should we not align this?
so moving the defaulting also to config here?

Reply:
Thanks for bringin it up!
This would mean that the `.ui5rc` file would always receive a value for the `ui5DataDir` and this setting would always be in there, regardless of the fact it's set intentionally or not.
Do we want to always have this in the `.ui5rc` file?

Also, the resolvers (i.e. `Openui5Resolver.js`) that utilize this directory are exposed through the `package.json` and in the best case we would need to double maintain the `ui5HomeDir`- in the `AbstractResolver.js` and in the `Configuration.js`. 
We cannot delete the default value from the `AbstractResolver.js (-> Openui5Resolver.js)` where it's populated currently, because that would be an incompatible change.

What do you think? Which way to choose? :D

Reply:
@RandomByte any opinion on the preferred way?

Reply:
Good question. I think we discussed this as part of [RFC 0013 Configuration](https://github.com/SAP/ui5-tooling/blob/main/rfcs/0013-configuration.md#configuration-module)

There it says:
> [...] **Parameters shall not have default values.** [...]

So I think we should not add a defaulting here.

This means that Configuration.js is not the single source of truth for which ui5DataDir is used, but that's our compromise. Every consumer needs to apply the defaulting themselves, or we add a dedicated class/function somewhere to centralize that.

Actually, the environment variable must also not be read and applied in Configuration.js but on the consumer side. As per the same RFC:
> There shall **not be any fallback mechanism to environment variables** or similar inside the Configuration.js module. [...]

Reply:
Ah, thanks a lot for clearing this out!